src/peergos/shared/storage/Transaction.java

```diff
@@ -0,0 +1,33 @@
+package peergos.shared.storage;
+
+import peergos.shared.crypto.hash.*;
+
+import java.util.concurrent.*;
+import java.util.function.*;
+
+public class Transaction {
+
+    /** Run a series of operations under a transaction, ensuring that it is closed correctly
+     *
+     * @param owner
+     * @param processor
+     * @param ipfs
+     * @param <V>
+     * @return
+     */
+    public static <V> CompletableFuture<V> run(PublicKeyHash owner,
```

Comment:
nit about  the name: since it has a non-void return value can we name it  ``call``?

Reply:
Good idea. I've done it in the last PR to avoid rebasing.